### ISSUE TITLE
repoOverrides when using flake

### ISSUE THREAD

**Issue by moritzschaefer** [nix-community/NUR#254]
I am using flake for my nixos configuration and was able to add NUR as described in [the README](https://github.com/nix-community/NUR#flake-support).

However, as I am starting my own NUR now, and want to test packages locally first, I would like to use the `repoOverrides` attribute to add my local NUR repository. 

Unfortunately, I don't see a way to pass the `repoOverrides` attribute to NUR's default.nix when using flakes.

Is it possible to support this feature?

**Comment by Mic92** [comment]
Not yet. This would be nice to have. I would say flake support is not set in stone yet. There might be better ways how this could be integrated into NUR. After all we could just have an open registry like https://github.com/NixOS/flake-registry

**Comment by augu5te** [comment]
By transposing the initial "Overriding repositories" below the ways  I obtained with **packageOverrides** or **overlay** :

- with packageOverrides 
```nix
  inputs.nur.url = "github:nix-community/NUR";
  inputs.kapack.url = "path:/home/auguste/dev/nur-kapack";

  outputs = {self, nixpkgs, nur, kapack }: {
 
  nurpkgs = import nixpkgs { inherit system; };

  system = "x86_64-linux";
 ...
  modules = [
       {
         nixpkgs.config.packageOverrides = pkgs: {
            nur = import nur {
              inherit pkgs nurpkgs;
              repoOverrides = { kapack = import kapack { inherit pkgs; }; };
            };
          };
        }
  ];
...
```
- with overlay
```nix
 ...
 modules = [
 {
    nixpkgs.overlays = [
      (final: prev: {
        nur = import nur {
          nurpkgs = prev;
          pkgs = prev;
          repoOverrides = { kapack = import kapack { pkgs = prev; }; };
        };
      })
    ];
   } 
 ...
```

**Comment by Mic92** [comment]
We could also have these snippets in the README.

**Comment by milahu** [comment]
> ```nix
> {
>   inputs.kapack.url = "path:/home/auguste/dev/nur-kapack";
> ```

this breaks when the last component of the path is a symlink

```nix
{
  # last component is symlink
  # $ readlink /home/user/src/symlink-to-nur-packages
  # nur-packages
  # $ sudo nixos-rebuild switch
  # error: getting status of '/nix/store/nur-packages': No such file or directory
  inputs.nur-packages-local.url = "path:/home/user/src/symlink-to-nur-packages";

  # ok
  #inputs.nur-packages-local.url = "path:/home/user/src/nur-packages";
```

may be fixed by https://github.com/NixOS/nix/pull/6530

**Comment by milahu** [comment]
```diff
-  inputs.nur-packages-local.url = "path:/home/user/src/nur-packages"
+  inputs.nur-packages-local.url = "git+file:///home/user/src/nur-packages"
```

is better, because it will copy only files tracked by git

**Comment by aacebedo** [comment]
Is this solution still working? I I tried to apply it and facing an error.
Posted tgi question with details in the discourse https://discourse.nixos.org/t/cannot-use-nur-modules-in-nixos-flake/49402
and in the NUR element channel.

**Comment by augu5te** [comment]
Yes I works have a look to:  https://github.com/oar-team/regale-nixos-compose/blob/main/bebida/flake.nix

**Comment by aacebedo** [comment]
> Yes I works have a look to: https://github.com/oar-team/regale-nixos-compose/blob/main/bebida/flake.nix

Thanks for the answer. 
However the integration in the link is 
not done the same way as in the Readme, I don't see the overlays.
